lxd/instances_post.go

```diff
@@ -377,7 +388,8 @@ func createFromMigration(s *state.State, r *http.Request, projectName string, pr
 
 		instOp.Done(nil) // Complete operation that was created earlier, to release lock.
 		runRevert.Success()
-		return nil
+
+		return instanceCreateFinish(s, req, args)
```

Comment:
So I think this has the potential to cause problems when doing a stateful (live) migration, as the instance is naturally restored on the target in a running (started) state, so if you request it be started it will fail the migration because its already running.

Please can you running LXD in 2 VMs, add a remote so they can talk to it other using `lxc remote add` and then try live migrating a nested VM (a VM running inside one of the LXD VMs) to the other run, but with this new `Start` field set to `true` and see if it breaks.

If you need help setting this up let me know and we can arrange a 1:1 video call to go through it.

Reply:
As you expected, this caused failure for stateful migration. I added a check in `instanceCreateFinish` to ensure we do not attempt to start a container that is already running.

Reply:
thanks! Let me know when ready for review.